src/client.ts

```diff
@@ -117,54 +120,47 @@ export class SymplifySDK {
             return null;
         }
 
+        const visitorID = ensureVisitorID(siteData, this.idGenerator);
+        if (!visitorID) {
+            this.log.error("could not get or generate a visitor ID");
+            return null;
+        }
+
         const project = findProjectWithName(this.config.latest, projectName);
         if (!project) {
             this.log.error(`findVariation: unknown project: ${projectName}`);
             return null;
         }
 
+        // 1. if previewing a project, handle and return early
+
+        if (siteData.getPreviewData() !== null) {
+            return handlePreview(siteData, project, cookies, audienceAttributes, this.log);
+        }
+
         if (project.state !== "active") {
             return null;
         }
 
-        const currAllocation = siteData.getAllocation(project);
-
-        switch (currAllocation) {
-            case undefined:
-                // no allocation data
-                break;
-            case null:
-                // explicit null allocation
-                return null;
-            default:
-                // variation allocation
-                return currAllocation.name;
+        // 2. if we already have an allocation from a previous visit, use that and return early
+
+        if (siteData.getAllocation(project) !== undefined) {
```

Comment:
Same as the switch from before, if not undefined, there is allocation info in the cookie (and that can be null or some variation). If it's undefined we fall through to the new allocation logic.